metricflow/specs/patterns/spec_pattern.py

```diff
@@ -0,0 +1,22 @@
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+from typing import Sequence
+
+from metricflow.specs.specs import LinkableInstanceSpec
+
+
+class SpecPattern(ABC):
```

Comment:
nit: LinkableSpecPattern? It only works on LinkableInstanceSpecs.

Reply:
Actually, this was updated later in the stack to apply for any spec. Moved those changes into this commit.